src/sentry_tracing.c

```diff
@@ -205,6 +205,27 @@ sentry__transaction_decref(sentry_transaction_t *tx)
     };
 }
 
+void
+sentry__span_incref(sentry_span_t *span)
+{
+    sentry_value_incref(span->inner);
+}
+
+void
+sentry__span_decref(sentry_span_t *span)
+{
+    if (!span) {
+        return;
+    }
+
+    if (sentry_value_refcount(span->inner) <= 1) {
+        sentry_value_decref(span->inner);
+        sentry_free(span);
```

Comment:
👍🏻 I find it quite interesting to piggyback on the inner values refcount.

Reply:
That wasn't me, these functions are literally copy-pasted from the corresponding functions for transactions ^^

Reply:
This was a quick and easy way to staple refcounting onto these types without going the full mile, but it could be wrong in some way; I'll admit that I'm not too good at this C thing. Could there be subtle issues with this approach?

For example, there's a small issue with the above ref counting implementation lead to the creation of `sentry__span_free` (L268-280 in the original, or L289-301 in this PR) instead of `sentry__span_incref` and `sentry__span_decref`: if we pass ownership of the `inner` `sentry_value_t` to somebody else, `sentry__span_decref` is unable to determine on its own whether it is supposed to free the wrapping span.

Reply:
Having thought about it a little, I think I have an idea on how to deal with a very specific instance of the example that exists in the SDK, but the general problem is still there.

I'll see what I can do to implement the fix.